src/components/header/movie-details/MovieDetails.tsx

```diff
@@ -1,25 +1,35 @@
 import "./MovieDetails.css";
-import {Movie} from "../../models/movies";
 import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
 import {faSearch} from '@fortawesome/free-solid-svg-icons';
 import React from "react";
+import useMovieById from "../../../hooks/UseMovieById";
+import {useParams} from "react-router-dom";
+import useMultipleSearchParams from "../../../hooks/UseMultipleSearchParams";
 
-function MovieDetails({movie, goBackClick} : {movie: Movie | null, goBackClick: () => void}): React.JSX.Element {
+function MovieDetails(): React.JSX.Element {
+    const params = useParams();
+    const movieId = params.movieId ?  Number(params.movieId) : null;
+    const movie = useMovieById(movieId);
+    const { updateQueryParams } = useMultipleSearchParams();
     const convertToHours = (totalMinutes: number) => {
         const hours = Math.floor(totalMinutes / 60);
         const minutes = totalMinutes % 60;
 
         return `${hours}h ${minutes}min`;
     }
 
+    const goBack = () => {
+        updateQueryParams({}, '/');
```

Comment:
You could return 'navigate' from your hook. And just use it here, as it should be used